packages/locale/src/format/cldr/CldrCache.js

```diff
@@ -0,0 +1,54 @@
+/*
+* @copyright (c) 2020, Philipp Thuerwaechter & Pattrick Hueper
+* @license BSD-3-Clause (see LICENSE.md in the root directory of this source tree)
+*/
+
+import cldrData from 'cldr-data';
+import Cldr from 'cldrjs';
+
+const cldrDataLoaded = new Set();
+/**
+* Loads the Cldr data for the given path if it hasn't been loaded before.
+*/
+export const loadCldrData = (path) => {
+    if (!cldrDataLoaded.has(path)) {
+        Cldr.load(cldrData(path));
+        cldrDataLoaded.add(path);
+    }
+};
+
+const localeToCldrInstanceCache = {};
```

Comment:
I _think_ each Cldr instance is fairly small memory-wise (since I imagine it just holds a few references to the already loaded Cldr data and doesn't copy it or anything) so this cache seems fairly harmless memory-wise to me.

Cldr instances do [have some sort of per-instance event listener system](https://github.com/rxaviers/cldrjs#cldreventjs), but I don't think that's being used anywhere, so sharing a single instance per locale string should be fine runtime-behavior-wise.

Reply:
> I wonder why you use an object here and for the other cache a new Map(), any reason for that ?

Not any great reason - I was just generally using a `Map` where forced to because of using objects as keys, and using a regular object everywhere else.

> About cache size, i am unsure, i don't know what the Cldr is storing internally. Could we somehow measure memory usage ? Like write a test with a typical scenarios and capture memory usage, so we could get an idea, how much this could get.

I'm not sure either how much extra memory each instance uses. I think doing a quick test to try to verify that makes sense, in case it is a lot of copied data per instance.

Reply:
I wrote a loop in some test initialization code that just created 100 Cldr instances, and then used the Chrome allocation recorder to measure what was allocated before and after that loop. It looks like the Cldr instances are fairly harmless, a little less than a KB each (see screenshot below), so it seems that keeping them around should be fine.

I'm not 100% confident that this was an accurate test, just because I'm not too familiar with the tools around measuring memory allocations, but as far as I can tell, each instance is pretty light.

![image](https://user-images.githubusercontent.com/1400151/88469060-96f37d00-ceba-11ea-98a2-95ca41063a93.png)

Reply:
LGTM, so the the Cldr instance from external code seems to be harmless.